packages/knip/src/plugins/astro/index.ts

```diff
@@ -19,6 +23,11 @@ const production = [
   'src/actions/index.{js,ts}',
 ];
 
+const resolveFromAST: ResolveFromAST = (sourceFile: ts.SourceFile) => {
+  const componentPaths = getComponentPathsFromSourceFile(sourceFile);
+  return [...entry.map(toEntry), ...[...production, ...componentPaths].map(id => toProductionEntry(id))];
```

Comment:
A few places where we are spreading into one array. Any chance we could have duplicate paths in the resulting array? I wonder if it's worth deduping this before returning.

Reply:
Eventually things should be deduped before the actual globbing or reporting takes place.